### PR TITLE
Support presets in AutoMM predictor initialization

### PR DESCRIPTION
**Pull request by zhiqiangdon**
*Issue #, if available:*

*Description of changes:*
1. Let users use `presets` in predictor initialization. The presets is to control model quality: `best_quality`, `high_quality_fast_inference`, and `medium_quality_faster_inference`.
2. Fix ner presets. Use the registered ner presets rather than the default.
3. Add unit tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/presets.py [hunk 1]
```diff
@@ -465,19 +471,36 @@ def best_quality_ner():
...
+            "timm_image",
+            "ner_text",
+            "fusion_ner",
+        ],
         "model.ner_text.checkpoint_name": "microsoft/deberta-v3-base",
     }
 
 
+@automm_presets.register()
+def ner():
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/presets.py:500]: Shall we call it `default_ner`?

**Comment by sxjscience** [comment]
LGTM in general. Minor comment on the naming.

We can consider to register `default_{problem_type}` for all the problem types, rather than using `{problem_type}`.

File: multimodal/src/autogluon/multimodal/presets.py (hunk 1 above)
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/presets.py:500]: Currently, all specific problem types, e.g., `image_text_similarity` and `object_detection`, have registered presets. `default` is for the general problem types `binary`, `multiclass`, and `regression`.

**Review by bryanyzhu — APPROVED** [review]
Looks good. One side question, where should we introduce these presets to users, in the apis documentation or automm customization tutorial page?

**Pull request merged**